nca/NetworkConfig/NetworkConfigQuery.py

```diff
@@ -1131,27 +1413,47 @@ def exec(self, only_captured=False):
 
         if not_contained_list:
-            return self._query_answer_with_relevant_explanation(sorted(not_contained_list))
+            return QueryAnswer(False, negative_output_result, sorted(not_contained_list))
+
+        return QueryAnswer(True, self.name1 + ' is contained in ' + self.name2, numerical_result=1)
 
-        output_result = self.name1 + ' is contained in ' + self.name2
-        return QueryAnswer(True, output_result, numerical_result=1)
+    def execute_and_write_output_in_required_format(self, cmd_line_flag=False):
+        query_answer = self.exec()
+        query_result = query_answer.numerical_result if not cmd_line_flag else not query_answer.bool_result
+        explanation_header = ''
+        yaml_description = []
+        if query_answer.output_explanation:
+            assert isinstance(query_answer.output_explanation, list)
+            explanation_header = \
+                self.determine_explanation_header(self.output_config.configName, self.output_config.secondConfigName,
+                                                  query_answer.output_explanation[0])
+            yaml_description = [explanation_header]
 
-    def _query_answer_with_relevant_explanation(self, explanation_list):
-        output_result = f'{self.name1} is not contained in {self.name2}'
-        output_explanation = f'\nConnections allowed in {self.name1} which are not a subset of those in {self.name2}:\n'
-        output_explanation += '\n'.join(explanation_list)
-        return QueryAnswer(False, output_result, output_explanation)
+        query_output = self.write_query_output(query_answer, explanation_header, yaml_description, int(query_result))
+        return query_result, query_output, query_answer.query_not_executed
 
     @staticmethod
-    def compute_query_output(query_answer, cmd_line_flag=False):
-        res = query_answer.numerical_result if not cmd_line_flag else not query_answer.bool_result
-        query_output = query_answer.output_result + query_answer.output_explanation + '\n'
-        return res, query_output, query_answer.query_not_executed
+    def determine_explanation_header(config_name_1, config_name_2, first_elem):
+        """
+        determines the explanation header message according to the type of elements in
+        the query_answer.output_explanation list
+        The output explanation of the ContainmentQuery may be :
+        (1) list[str] - when configs containment is not fulfilled because of different peers
+        (2) list[tuple] - when configs containment is not fulfilled because of different connections between same peers
+        :param str config_name_1: the name of the first config
+        :param str config_name_2: the name of the second config
+        :param first_elem: the first element of the output_explanation to determine its type
+        :rtype: str
+        """
+        if isinstance(first_elem, str):
```

Comment:
I would consider changing the numerical result of this query to determine if it is not contained due to pods or actual connections. (such as 0 for not contained due to connections, 1 for containment, 2 for not contained due to pods).
So we don't have to rely on the type of the elements in the output explanation to determine this.